packages/block-editor/src/components/content-only-controls/index.js

```diff
@@ -42,22 +42,59 @@ function BlockAttributeToolsPanelItem( {
 	attributeValues,
 } ) {
 	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+	const { selectBlock, toggleBlockSpotlight } = unlock(
+		useDispatch( blockEditorStore )
+	);
+	const { isBlockSelected } = useSelect( blockEditorStore );
 	const ControlComponent = controls[ control.type ];
 
 	if ( ! ControlComponent ) {
 		return null;
 	}
 
-	return (
-		<ControlComponent
-			clientId={ clientId }
-			control={ control }
-			blockType={ blockType }
-			attributeValues={ attributeValues }
-			updateAttributes={ ( attributes ) =>
-				updateBlockAttributes( clientId, attributes )
+	const isSelected = isBlockSelected( clientId );
```

Comment:
Bit of a nit, but this line stands out to me a little bit because it's not within a `useSelect` where we'd usually expect. Is that because you wanted to call it after the check for whether there's a `ControlComponent`? If so, one option could be to move the check for `controls[ control.type ]` out of this function and to the `fields?.map` in `BlockFields` instead, then the check to see if the block is selected could be in the `useSelect`?

Reply:
Good idea, i'll do the check in the map 👍🏻